### PR TITLE
Support using OPS token from environment variable

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#347501](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/347501)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6884)

### CODE REVIEW

File: src/docfx/config/EnvironmentVariable.cs [hunk 1]
```diff
@@ -19,6 +19,8 @@ public static class EnvironmentVariable
 
         public static string? CorrelationId => GetValue("DOCFX_CORRELATION_ID");
 
+        public static string? OPSBuildUserToken => GetValue("DOCFX_OPS_BUILD_USER_TOKEN");
```
Comment (yufeih) [inline src/docfx/config/EnvironmentVariable.cs:22]:
```suggestion
        public static string? OPSBuildUserToken => GetValue("DOCFX_OPS_TOKEN");
```

File: src/docfx/config/ops/OpsAccessor.cs [hunk 2]
```diff
@@ -334,14 +334,20 @@ private static async Task FillOpsToken(HttpRequestMessage request, DocsEnvironme
...
+                    var token = EnvironmentVariable.OPSBuildUserToken;
+                    if (string.IsNullOrEmpty(token))
                     {
-                        DocsEnvironment.Prod => s_opsTokenProd,
-                        DocsEnvironment.PPE => s_opsTokenSandbox,
-                        _ => throw new InvalidOperationException(),
-                    };
-
-                    request.Headers.Add("X-OP-BuildUserToken", await token.Value);
+#pragma warning disable SA1119 // Statement should not use unnecessary parenthesis
```
Comment (yufeih) [inline src/docfx/config/ops/OpsAccessor.cs:340]: why disable SA1119 here?

File: src/docfx/config/ops/OpsAccessor.cs [hunk 3]
```diff
@@ -334,14 +334,20 @@ private static async Task FillOpsToken(HttpRequestMessage request, DocsEnvironme
                 // For development usage
                 try
                 {
-                    var token = (environment ?? DocsEnvironment) switch
+                    var token = EnvironmentVariable.OPSBuildUserToken;
+                    if (string.IsNullOrEmpty(token))
```
Comment (yufeih) [inline src/docfx/config/ops/OpsAccessor.cs:338]: use `??`

File: src/docfx/config/ops/OpsAccessor.cs (hunk 2 above)
Comment (928PJY) [reply src/docfx/config/ops/OpsAccessor.cs:340]: seems it doesn't report rightly

File: src/docfx/config/ops/OpsAccessor.cs (hunk 3 above)
Comment (928PJY) [reply src/docfx/config/ops/OpsAccessor.cs:338]: seems ?? don't handle the empty string?

Comment (yufeih) [reply src/docfx/config/ops/OpsAccessor.cs:338]: `EnvironmentVariable.GetValue` internally handles empty string, to make it work with `??`

**Review by yufeih — APPROVED** [review]

File: src/docfx/config/ops/OpsAccessor.cs (hunk 3 above)
Comment (928PJY) [reply src/docfx/config/ops/OpsAccessor.cs:338]: Hah, great, but seem it cannot be used in this case since we need another variable `tokenTask`

**Pull request merged**